daemon/emer-aggregate-tally.c

```diff
@@ -178,16 +187,19 @@ delete_tally_entries (EmerAggregateTally  *self,
     }
   g_string_append (query, ");");
 
-  if (!CHECK (sqlite3_prepare_v2 (self->db, query->str, -1, &stmt, NULL)) ||
-      !CHECK (sqlite3_step (stmt)) ||
-      !CHECK (sqlite3_reset (stmt)) ||
-      !CHECK (sqlite3_finalize (stmt)))
-    {
-      g_prefix_error (error, "Failed to delete %u tally entries: ", rows_to_delete->len);
-      return FALSE;
-    }
+  gboolean ret =
+      CHECK (sqlite3_prepare_v2 (self->db, query->str, -1, &stmt, NULL)) &&
+      CHECK (sqlite3_step (stmt));
 
-  return TRUE;
+  if (ret)
+    ret = CHECK (sqlite3_finalize (stmt));
+  else
+    sqlite3_finalize (stmt);
```

Comment:
This repeated pattern feels really ugly but I can't think of a way to make it more concise without adding yet more macros. The `CHECK` macro is already a little bit too magic for my tastes today – and I wrote it!

Reply:
I tried writing another macro but it actually made it worse.